Mathlib/Algebra/MvPolynomial/Nilpotent.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.MvPolynomial.Homogeneous
+import Mathlib.RingTheory.Polynomial.Nilpotent
+
+/-!
+# Nilpotents and units in multivariate polynomial rings
+
+We prove that
+- `MvPolynomial.isNilpotent_iff`:
+  A multivariate polynomial is nilpotent iff all its coefficents are.
+- `MvPolynomial.isUnit_iff`:
+  A multivariate polynomial is invertible iff its constant is invertible
+  and its other coefficients are nilpotent.
+-/
+
+@[simps]
+noncomputable
+def Finsupp.optionEquiv {α M : Type*} [Zero M] : (Option α →₀ M) ≃ M × (α →₀ M) where
+  toFun P := (P .none, P.some)
+  invFun P := (P.2.embDomain .some).update .none P.1
+  left_inv P := by ext (_|a) <;> simp [Finsupp.update]; simp [← Function.Embedding.some_apply]
+  right_inv P := by ext <;> simp [Finsupp.update]; simp [← Function.Embedding.some_apply]
+
+@[simp]
+lemma Finsupp.some_update_none {α M : Type*} [Zero M] (P : Option α →₀ M) (a : M) :
```

Comment:
Does this lemma help with the equiv above if you swap the order?

Reply:
Yeah I've already cleaned that up and moved them to the right file.